app/controllers/clone_pages_controller.rb

```diff
@@ -0,0 +1,19 @@
+class ClonePagesController < ApplicationController
+  before_action :authenticate_user!
+
+  def new
+    page
+  end
+
+  def create
+    new_page = PageCloner.clone(page, params[:page][:title])
+    QueueManager.push(new_page, job_type: :create)
+    redirect_to edit_page_path(new_page)
```

Comment:
does this redirect to the url with the page id or the page slug? page slug is def more desirable